utils/removeDiscordRoleFromUser.js

```diff
@@ -0,0 +1,44 @@
+const discordRolesModel = require("../models/discordactions");
+const discordServices = require("../services/discordService");
+const { logType } = require("../constants/logs");
+const { addLog } = require("../models/logs");
+
+/**
+ * Removes a Discord role from a user using Discord Id.
+ *
+ * @param {Object} userData - User data.
+ * @param {string} discordId - User's Discord ID.
+ * @param {string} roleid - Discord Role ID.
+ *
+ * @returns {Promise<Object>} - Result with success status and message.
+ */
+export const removeDiscordRoleFromUser = async (userData, discordId, roleid) => {
+  try {
+    const role = await discordRolesModel.isGroupRoleExists({ roleid });
+
+    if (!role.roleExists) {
+      throw new Error("Role doesn't exist");
+    }
+
+    const roleData = role.existingRoles.docs[0].data();
```

Comment:
why are we doing this? We already have access to roleId.

Reply:
This is a common function. Why should we assume that anyone using this function will be passing correct roleid through params?

Reply:
I pointed to line 23.

Reply:
Oh, yes you're right. we already have access to roleid but doing this to make sure that we are using the correct roleid only returned directly from db and not relying on external params. If we didn't have that we have surely used the one coming from params. Now if we are getting the id generated through process. So I thought it would be better to use that.

Reply:
You already have a condition for checking role.

Reply:
Okay made the necessary changes that you asked. Please have a look.